kpconv_torch/test.py

```diff
@@ -26,72 +27,38 @@
     Toronto3DSampler,
 )
 from kpconv_torch.models.architectures import KPCNN, KPFCNN
-from kpconv_torch.utils.config import Config
 from kpconv_torch.utils.tester import ModelTester, get_test_save_path
 
 
-def model_choice(chosen_log):
-    ###########################
-    # Call the test initializer
-    ###########################
-    # Automatically retrieve the last trained model
-    if chosen_log in ["last_ModelNet40", "last_ShapeNetPart", "last_S3DIS"]:
-
-        # Dataset name
-        test_dataset = "_".join(chosen_log.split("_")[1:])
-
-        # List all training logs
-        logs = np.sort(
-            [os.path.join("results", f) for f in os.listdir("results") if f.startswith("Log")]
-        )
-
-        # Find the last log of asked dataset
-        for log in logs[::-1]:
-            log_config = Config()
-            log_config.load(log)
-            if log_config.dataset.startswith(test_dataset):
-                chosen_log = log
-                break
-
-        if chosen_log in ["last_ModelNet40", "last_ShapeNetPart", "last_S3DIS"]:
-            raise ValueError('No log of the dataset "' + test_dataset + '" found')
-
-    # Check if log exists
-    if not os.path.exists(chosen_log):
-        raise ValueError("The given log does not exists: " + chosen_log)
-
-    return chosen_log
-
-
 def main(args):
     test(
         args.datapath,
+        args.configfile,
         args.filename,
         args.chosen_log,
-        args.dataset,
-        args.n_votes,
-        args.validation_size,
-        args.potential_increment,
     )
 
 
 def test(
     datapath: Path,
+    configfile: Path,
     filename: str,
     chosen_log: Path,
-    dataset: str,
-    n_votes: int = 100,
-    validation_size: int = None,
-    potential_increment: int = None,
 ) -> None:
+
     # Choose the index of the checkpoint to load OR None if you want to load the current checkpoint
     chkp_idx = -1
 
-    # Choose to test on validation or test split
+    # Choose to test on validation or test task
     on_val = True
 
     # Deal with 'last_XXXXXX' choices
-    chosen_log = str(model_choice(chosen_log))
+    output_path = get_test_save_path(filename, chosen_log)
+    if configfile is not None:
+        config_file_path = configfile
+    else:
+        config_file_path = Path(chosen_log / "config.yml")
```

Comment:
1. I made the choice of removing the `model_choice` function, made to find back the last trained model, in case the user asks for it from the CLI. Since the dataset is now passed to the command through the config file and not through the `-s` parameter anymore, this mechanism is obsolete. The means to find it back would be to accept, from inside the config file, to ask for `last_S3DIS` instead of `S3DIS`. As a consequence, the `-l` parameter would become useless (as it was already the cas in the former version). I did not find this way of defining the CLI very logical, so I removed this possibility and supposed that the user would ask the last one, through the correct specification of the `-l` parameter. 

2. The `if... else` structure was added following one of our discussions on a review. It is intended to distinguish the case when the inference is done with a specific config file and the case when you want to use the config file already created by the training of the model. It was absolutely necessary before I created a "test" section in the config file. Now maybe, it will be rarely used. I don't know if it can happen that the user want another config file than the one generated by the training of the model.

What do you think?